File: app/src/main/java/com/elegion/android/ui/base/view/NoInternetStubView.java
package com.elegion.android.ui.base.view;

import com.arellomobile.mvp.viewstate.strategy.StateStrategyType;
import com.elegion.android.util.moxy.strategy.AddToEndSingleByTagStrategy;
import com.elegion.android.util.moxy.strategy.SkipByTagStrategy;

/**
 * @author mikhail barannikov
 */
public interface NoInternetStubView {
    @StateStrategyType(value = AddToEndSingleByTagStrategy.class, tag = "NoInternetStubView")
    void showNoInternetStub();

    @StateStrategyType(value = SkipByTagStrategy.class, tag = "NoInternetStubView")
    void hideNoInternetStub();
}

File: app/src/main/kotlin/com/elegion/android/bankex/ui/base/view/NoInternetStubView.kt
package com.elegion.android.bankex.ui.base.view

import com.arellomobile.mvp.viewstate.strategy.StateStrategyType
import com.elegion.android.bankex.util.moxy.strategy.AddToEndSingleByTagStrategy
import com.elegion.android.bankex.util.moxy.strategy.SkipByTagStrategy

interface NoInternetStubView {
    @StateStrategyType(value = AddToEndSingleByTagStrategy::class, tag = "NoInternetStubView")
    fun showNoInternetStub()

    @StateStrategyType(value = SkipByTagStrategy::class, tag = "NoInternetStubView")
    fun hideNoInternetStub()
}
